ethportal-peertest/src/scenarios/offer_accept.rs

```diff
@@ -68,3 +72,55 @@ pub fn test_offer_accept(peertest_config: PeertestConfig, peertest: &Peertest) {
         "The received content {received_content_str}, must match the expected {HISTORY_CONTENT_VALUE}",
     );
 }
+
+pub async fn test_populate_offer(peertest_config: PeertestConfig, peertest: &Peertest) {
+    info!("Testing Populate Offer/ACCEPT flow");
+
+    let ipc_client = reth_ipc::client::IpcClientBuilder::default()
+        .build(&peertest_config.target_ipc_path)
+        .await
+        .unwrap();
+
+    // Offer unique content key to bootnode
+    let content_key: HistoryContentKey = serde_json::from_value(json!(
+        "0x00cb5cab7266694daa0d28cbf40496c08dd30bf732c41e0455e7ad389c10d79f4f"
+    ))
+    .unwrap();
+    let content_value: HistoryContentItem =
+        serde_json::from_value(json!(HISTORY_CONTENT_VALUE)).unwrap();
+
+    let result = ipc_client
+        .offer(
+            Enr::from_str(&peertest.bootnode.enr.to_base64()).unwrap(),
+            content_key,
+            Some(content_value.clone()),
+        )
+        .await
+        .unwrap();
+
+    // Check that ACCEPT response sent by bootnode accepted the offered content
+    validate_portal_offer(&serde_json::to_value(result).unwrap(), peertest);
+
+    // Check if the stored content item in bootnode's DB matches the offered
+    let mut received_content_value = ipc_client
+        .local_content(serde_json::from_value(json!(HISTORY_CONTENT_KEY)).unwrap())
+        .await
+        .unwrap();
+
+    let mut counter = 0;
+    while received_content_value == HistoryContentItem::Unknown("".to_owned()) && counter < 5 {
+        error!("Retrying after 0.5sec, because content should have been present");
+        thread::sleep(time::Duration::from_millis(500));
+        received_content_value = ipc_client
+            .local_content(serde_json::from_value(json!(HISTORY_CONTENT_KEY)).unwrap())
+            .await
+            .unwrap();
+
+        counter += 1;
+    }
+
+    assert_eq!(
+        content_value, received_content_value,
+        "The received content {received_content_value:?}, must match the expected {HISTORY_CONTENT_VALUE}",
```

Comment:
I'm assuming it doesn't make much of a difference in the displayed value, but I'd update `HISTORY_CONTENT_VALUE` -> `content_value` to avoid slight confusion